src/Source Projects/MSX Applications/MSX ROM Application/Make/make.sh

```diff
@@ -87,45 +91,45 @@ compile () {
     echo -----------------------------------------------------------------------------------
     echo Building application modules...
     
-    OBJLIST=
+    let I=0
     while read -r APPFILE; do
         if [[ -n $APPFILE && ${APPFILE:0:1} != ';' ]]; then
             APPFILE=$(path_replace "$APPFILE" '[MSX_LIB_PATH]' "$MSX_LIB_PATH")
             APPFILE=$(path_replace "$APPFILE" '[MSX_OBJ_PATH]' "$MSX_OBJ_PATH")
             RELFILE="$MSX_OBJ_PATH"/$(basename "${APPFILE/.*/}").rel
             FILEEXT=$(basename "${APPFILE/*./}")
             if [[ ".$FILEEXT" == '.c' ]]; then
-                echo -n "Processing C file $(basename "$APPFILE")... "
-                TMPFILE=$(mktemp $(basename "$RELFILE.XXXXX"))
-                eval "sdcc -mz80 -c $INCDIRS -o $RELFILE $APPFILE 2> $TMPFILE"
+                echo "Processing C file $(basename "$APPFILE")... "
+                OUTPUT=$(_exec Make/sdcc -mz80 -c ${INCDIRS[*]} -o "'$RELFILE'" "'$APPFILE'")
             else
-                echo -n "Processing ASM file $(basename "$APPFILE")... "
-                TMPFILE=$(mktemp $(basename "$RELFILE.XXXXX"))
-                sdasz80 -o "$RELFILE" "$APPFILE" 2> "$TMPFILE"
+                echo "Processing ASM file $(basename "$APPFILE")... "
+                OUTPUT=$(_exec Make/sdasz80 -o "'$RELFILE'" "'$APPFILE'")
```

Comment:
Instead of creating temporary file and deleting it later, we redirect output to variable. Less messy.